src/include/sof/lib/kcps-budget-manager.h

```diff
@@ -0,0 +1,26 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Krzysztof Frydryk <frydryk.krzysztof@intel.com>
+ */
+
+#ifndef __SOF_LIB_CPS_BUDGET_MANAGER_H__
+#define __SOF_LIB_CPS_BUDGET_MANAGER_H__
+
+#include <rtos/spinlock.h>
+#include <sof/sof.h>
+#include <stdint.h>
+
+struct kcps_budget_data {
+	/* uncache only */
+	int kcps_consumtion[CONFIG_CORE_COUNT];
+	struct k_spinlock lock;
+};
```

Comment:
I think this is ok with a comment. This has been a problematic approach if there is any need to access the data from multiple cores and performance is a concern (requiring cached access). The "kpcps_consumtion" items are here on the same cacheline, making this impossible to make multicore safe if anyone is accessing this data via a cached pointer.

Reply:
Performance shouldn't be a concern there. It's not a lot of data and should be used only when stuff is initializing or deinitializing.